README.md

```diff
@@ -1,2 +1,33 @@
 # hadoop-router-fs
-hadoop-router-fs provides an Hadoop [FileSystem](https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/fs/FileSystem.html) implementation that transforms URIs and routes file system operations between Hadoop file systems according to the path URI in the operation. 
+hadoop-router-fs provides an Hadoop [FileSystem](https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/fs/FileSystem.html) implementation that transforms URIs and routes file system operations between Hadoop file systems according to the path URI in the operation.
+
+RouterFS does a simple prefix replacement according to mapping configurations.
+
+## Mapping configuration structure
+
+Mapping configurations are Spark properties of the following form: 
+`routerfs.mapping.${toFsScheme}.${mappingIdx}.${replace/with}=${path-prefix}`
+
+Examples:
+```properties 
+routerfs.mapping.lakefs.1.replace=s3a://bucket/dir/ # mapping src
+routerfs.mapping.lakefs.1.with=lakefs://example-repo/dev/ # mapping dst
+routerfs.mapping.lakefs.2.replace=s3a://bucket/prefix # mapping src
+routerfs.mapping.lakefs.2.with=lakefs://example-repo/dev/prefix # mapping dst
+```
+Make sure your source prefix ends with a slash when needed. 
```

Comment:
When is a slash _not_ needed?  In particular, it seems always unsafe.  For instance, the above example will also change `s3a://bucket/prefixed` with `lakefs://example-repo/dev/prefixed`, which is surprising.